Makefile.core.mk

```diff
@@ -1,21 +1,37 @@
-export GO111MODULE ?= on
+# Copyright Istio Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
 
 lint:
 	@scripts/run_golangci.sh
-	@scripts/check_license.sh
+
+lint-buildifier:
 	@bazel run //:buildifier -- -showlog -mode=check $(git ls-files| grep -e BUILD -e WORKSPACE | grep -v vendor)
 
-.PHONY: testgrid
-testgrid:
-	configurator --prow-config prow/config.yaml --prow-job-config prow/cluster/jobs --output-yaml --yaml testgrid/default.yaml --oneshot --output testgrid/istio-generated.yaml
+test:
+	@go test -race ./...
+
+config-testgrid:
+	@GOARCH=amd64 GOOS=linux go get k8s.io/test-infra/testgrid/cmd/configurator@d5d7ce3eb0ffe35c899fe9358586cdffb6525899
+	@GOARCH=amd64 GOOS=linux go run k8s.io/test-infra/testgrid/cmd/configurator --prow-config prow/config.yaml --prow-job-config prow/cluster/jobs --output-yaml --yaml testgrid/default.yaml --oneshot --output testgrid/istio-generated.yaml
 
 generate-config:
-	(cd prow/config/cmd; go run generate.go write)
+	@(cd prow/config/cmd; GOARCH=amd64 GOOS=linux go run generate.go write)
 
 diff-config:
-	(cd prow/config/cmd; go run generate.go diff)
+	@(cd prow/config/cmd; GOARCH=amd64 GOOS=linux go run generate.go diff)
 
 check-config:
-	(cd prow/config/cmd; go run generate.go check)
+	@(cd prow/config/cmd; GOARCH=amd64 GOOS=linux go run generate.go check)
```

Comment:
Why do we need to force the architecture? Seems like this won't work for mac?

Reply:
We're building from a container. But we have a (I expect) temporary situation where we force the architecture of stuff built within the container to be the architecture of the host system. In this case, we need to build the binary in the container and run that binary, so we need to force the OS/ARCH to what the container needs.

We also want to support building outside of the container at which point this approach will not work. We'll need a few more variables to control this, but we're not there yet. So for now, we have a few spots across the repos where we force the OS/ARCH so that things work right in the container.